pkg/mapper/iac-providers/cft/config/cloudformation-stack.go

```diff
@@ -17,29 +17,47 @@
 package config
 
 import (
+	fn "github.com/accurics/terrascan/pkg/mapper/iac-providers/cft/functions"
 	"github.com/awslabs/goformation/v4/cloudformation/cloudformation"
 )
 
 // CloudFormationStackConfig holds config for aws_cloudformation_stack
 type CloudFormationStackConfig struct {
 	Config
-	TemplateURL      interface{} `json:"template_url"`
-	NotificationARNs interface{} `json:"notification_arns"`
+	TemplateURL      string            `json:"template_url"`
+	NotificationARNs interface{}       `json:"notification_arns"`
+	Parameters       map[string]string `json:"-"`
+	TemplateData     []byte            `json:"-"`
 }
 
 // GetCloudFormationStackConfig returns config for aws_cloudformation_stack
 func GetCloudFormationStackConfig(s *cloudformation.Stack) []AWSResourceConfig {
 	cf := CloudFormationStackConfig{
-		Config: Config{
-			Tags: s.Tags,
-		},
+		Config:           Config{Tags: s.Tags},
+		TemplateURL:      "",
+		NotificationARNs: nil,
+		TemplateData:     []byte{},
 	}
+
 	if len(s.NotificationARNs) > 0 {
 		cf.NotificationARNs = s.NotificationARNs
 	}
+
+	// Add and resolve template URL
 	if len(s.TemplateURL) > 0 {
 		cf.TemplateURL = s.TemplateURL
+
+		templateData, err := fn.DownloadBucketObj(s.TemplateURL)
+		if err == nil {
```

Comment:
any error that occurred during the download of the s3 object is lost and would not be captured in the `directory scan errors`. We should capture these errors in `directory scan errors` along with logging.

Reply:
Done this way to keep the violations in AWS::Cloudformation:: Stack intact. Propagating error will skip the violations for AWS::Cloudformation::Stack resource eveny time the nested template is not resolved.